lib/src/command/get.dart

```diff
@@ -60,6 +67,50 @@ class GetCommand extends PubCommand {
     );
   }
 
+  bool shouldRunPostGetHook() {
+    final flutterRoot = Platform.environment['FLUTTER_ROOT'];
+    if (flutterRoot == null) {
+      return false;
+    }
+    if (!fileExists(p.join(flutterRoot, 'version'))) {
+      return false;
+    }
+
+    // TODO: sky_engine check
+    final hasFlutterDependency =
+        entrypoint.root.dependencies.values.any((package) {
+      return package.name == 'flutter' &&
+          package.source.runtimeType == SdkSource;
+    });
+
+    return hasFlutterDependency;
+  }
+
+  Future<int> runPostGetHook() async {
+    final String flutterRoot = Platform.environment['FLUTTER_ROOT']!;
+    final String flutterToolPath = p.join(flutterRoot, 'bin', 'flutter');
+
+    final StreamSubscription<ProcessSignal> subscription =
+        ProcessSignal.sigint.watch().listen((e) {});
+    final Process process = await Process.start(
+      flutterToolPath,
+      [
+        'pub',
+        '_post_pub_get',
+        '-C',
+        directory,
+        '--update-version-and-package-config',
+        '--regenerate-platform-specific-tooling',
+        argResults['example'] ? '--example' : '',
+      ],
+      mode: ProcessStartMode.inheritStdio,
```

Comment:
Is there ever a need for the hook to be interactive?
I would prefer the hook to be run in `ProcessStartMode.normal`.